src/rust/engine/options/src/parse_tests.rs

```diff
@@ -648,3 +661,190 @@ fn test_parse_heterogeneous_dict() {
         )
     );
 }
+
+#[test]
```

Comment:
From here we have some very exhaustive tests of the fromfile mechanism. We also test this mechanism's application in args/env/config in the appropriate test files.

Reply:
Could we have a test or two covering some more error cases beyond just "not existing": the major class I can think of is when the file contents is invalid, both as a scalar, and, probably more interestingly, as a list or dict (e.g. where serde json/yaml can't even parse it).

Reply:
adding some list/dict cases as you suggest. There is no such thing as an invalid scalar at this level, since we don't convert to the type but just expand string to string. Can test for this in args/config/env maybe. I'll look into adding something.